app/src/lib/stores/app-store.ts

```diff
@@ -2514,6 +2525,22 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return repository
     }
 
+    let shouldPopStash = false
+
+    if (
+      currentBranch !== null &&
```

Comment:
this is a (non-blocking) nitpick, but could we pull this conditional into an enclosing `if` statement so its easier to follow that either `uncommittedChangesStrategy` will be acted on if `currentBranch !== null`